subsys/net/lib/openthread/platform/radio.c

```diff
@@ -337,11 +430,21 @@ void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
 otError otPlatRadioEnergyScan(otInstance *aInstance, u8_t aScanChannel,
 			      u16_t aScanDuration)
 {
-	ARG_UNUSED(aInstance);
-	ARG_UNUSED(aScanChannel);
-	ARG_UNUSED(aScanDuration);
+	energy_detection_time    = aScanDuration;
+	energy_detection_channel = aScanChannel;
 
-	return OT_ERROR_NOT_IMPLEMENTED;
+	clear_pending_events();
+
+	radio_api->set_channel(radio_dev, aScanChannel);
+
+	if (!radio_api->ed_scan(radio_dev, energy_detection_time,
+				energy_detected)) {
+		reset_pending_event(pending_events_detect_energy);
```

Comment:
Pending events were just cleared, is there a need to reset it again? Nonetheless, IMO it would be good to add LOG_ERR here in case we fail to setup the ED procedure.

Reply:
And return an appropriate error code of course.

Reply:
Ok I think I get it now. The problem is with `otPlatRadioEnergyScan` API specification - it simply does not accept the fact that triggering energy detection can fail. The only accepted return values are `OT_ERROR_NONE` and `OT_ERROR_NOT_IMPLEMENTED`. That said, IMO we could leave it as it is, but please add a comment here, why is this done like this.